### ISSUE TITLE
failure on "cudaStreamCreateWithFlags creation failed"

### ISSUE THREAD

**Issue by alinask** [openucx/ucx#2295]
The following is a failure which happends when runnning with the cuda transport.
Seems that it passes on two ranks but fails on more.

To reproduce on vulcan x2 (ppn=48(x2), nodelist=vulcan[01-02]):

module load hpcx-gcc-cuda 

mpirun -np 96 --display-map -mca btl self --tag-output --timestamp-output -mca pml ucx -mca coll '^hcoll' --bind-to hwthread -x UCX_NET_DEVICES=mlx5_0:1 -x UCX_IB_GID_INDEX=0 -mca osc ucx -x UCX_TLS=rc,cuda_copy,gdr_copy -mca opal_pmix_base_async_modex 0 -mca mpi_add_procs_cutoff 100000 --map-by node /hpc/mtr_scrap/users/mtt/scratch/ucx_ompi/20180210_171304_17497_141297_vulcan01/installs/74oZ/tests/hash/kmi_hash/tests/BENCH_QUERY -n 4000 -m 4000

```
Sat Feb 10 17:45:00 2018[1,88]<stdout>:[1518277500.381412] [vulcan01:14390:0] cuda_copy_iface.c:241  UCX  ERROR cudaStreamCreateWithFlags creation failed
Sat Feb 10 17:45:00 2018[1,70]<stdout>:[1518277500.399391] [vulcan01:14323:0] cuda_copy_iface.c:239  UCX  ERROR [cuda/cuda_copy/cuda_copy_iface.c:239] cuda failed with 2 
```

http://hpcweb.lab.mtl.com//hpc/mtr_scrap/users/mtt/scratch/ucx_ompi/20180210_171304_17497_141297_vulcan01/html/test_stdout_7LAwDP.txt

Full MTT report:
http://hpcweb.lab.mtl.com/hpc/mtr_scrap/users/mtt/scratch/ucx_ompi/20180210_171304_17497_141297_vulcan01/All_phase-summary.html

**Comment by bureddy** [comment]
cuda_copy uct is creating 2 cuda stream object in its iface init.  each cuda stream objects taking up huge memory(100MB).  So ppn 48 taken up all the memory on GPU (16GB) and some ranks are failed  to create stream object.

PR  https://github.com/openucx/ucx/pull/2308 fixes the issue by delayin the stream object creation on its first use.  this helps pure CPU applications not use any GPU resources.

**Comment by yosefe** [comment]
fixed by #2308 and #2310
